enhancements/cluster-logging/elasticsearch-v2-api-and-infrastructure.md

```diff
@@ -0,0 +1,319 @@
+  use up half of its memory for the JVM the other half is used for this. The less the JVM takes up the more it can map.
+  * The smaller a shard is, the less heap it can take up in responding to a query.
+  * If the data node is not coordinating, it does not need to compile the results of a query to then later return. This helps cut down on its heap usage.
+  * Elastic.co recommends also making data nodes do ingestion, so if there is an ingestion plugin that is used, it *may* require more memory
+
+* Coodinating nodes: These are the workers, they need enough memory so that they can build a query response and return it to the caller.
+  * Smaller shards and better formed queries along with index structure can help these use less memory.
+  * These will likely require the most memory of any other node though
+  * We match these 1:1 (at least) with a data node for communication efficiency (they would know more about shards that are on its data node)
+
+### Test Plan
+
+#### Unit Testing
+
+* We will need to add some more unit tests to confirm that we can generate objects with the same names when using V1 and new names when using V2
+
+#### Integration and E2E tests
+
+* A cluster can still be created using a V1 CR
+* A cluster can also be created using a V2 CR
+* A cluster that was created using V1 can be upgraded to V2 by creating the V2 with the same name (there is no possibility to downgrade)
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+* Operator is able to create clusters correctly and can scale them so that they are performant
+
+#### Tech Preview -> GA
+
+* Operator has shown consistent and reliable management of clusters that are highly performant
+* Operator is able to migrate V1 CRs to V2 CRs without data loss
+
+#### Removing a deprecated feature
+
+* We would seek to remove V1 roughly 2 releases after V2 is GA
+  * one release to account for information about V2 and announcing deprecation of V1
+  * one more release after that before deprecating
+
+### Upgrade / Downgrade Strategy
+
+* Only upgrade will be allowed from V1 to V2
+
+* Intended plan for upgrading would be:
+  * If there exists a V1 CR named "elasticsearch" in the namespace "openshift-logging"
+    and then a V2 CR named "elasticsearch" in the namespace "openshift-logging" is created then
+    the operator would ignore the V1 one and migrate the objects from V1 to V2 (objects primarily being currently used storage)
+
+  * The main concern with doing a straight migration (via transformation hook) is the need to try to extrapolate a desired t-shirt
+    size based on the current configurations which may be very difficult to get correct. It may also be that the customer's current
+    configuration does not correctly meet their demands so doing a migration could leave them with similar issues (mainly if their
+    storage capacity isn't high enough or if their cluster is under resourced)
```

Comment:
What about upgrading the tracing instances?

Reply:
there should be no difference. i simply gave "openshift-logging" as an example